pkg/termination/termination.go

```diff
@@ -127,58 +130,85 @@ func (h *handler) run(ctx context.Context) error {
 		}
 
 		// Instance not terminated yet
-		h.log.V(2).Info("Instance not marked for termination")
+		logger.V(2).Info("Instance not marked for termination")
 		return false, nil
 	}, ctx.Done()); err != nil {
 		return fmt.Errorf("error polling termination endpoint: %w", err)
 	}
 
-	if err := wait.PollImmediateUntil(h.pollInterval, func() (bool, error) {
-		h.log.V(1).Info("Getting node for machine")
-		machine, err := h.getMachineForNode(ctx)
-		if err != nil {
-			if errors.Is(err, notFoundMachineForNode{}) {
-				h.log.Error(err, "Machine not found for node")
-				return true, nil
-			}
-			h.log.Error(err, "error fetching machine for node:", h.nodeName)
-			return false, nil
-		}
+	// Will only get here if the termination endpoint returned FALSE
+	logger.V(1).Info("Instance marked for termination, marking Machine for deletion")
+	if err := h.markNodeForDeletion(ctx); err != nil {
+		return fmt.Errorf("error marking machine: %v", err)
```

Comment:
Should be node here ideally
```suggestion
	logger.V(1).Info("Instance marked for termination, marking Node for deletion")
	if err := h.markNodeForDeletion(ctx); err != nil {
		return fmt.Errorf("error marking node: %v", err)
```